### DISCUSSION TITLE
Telegram bot BusinessConnection

### DISCUSSION THREAD

**Discussion by IGNOR90** [chatwoot/chatwoot#9735] [Feature Requests]
### Is your feature or enhancement related to a problem? Please describe.

Telegram bot BusinessConnection 

### Describe the solution you'd like

https://core.telegram.org/bots/api#businessconnection
https://core.telegram.org/bots/features#bots-for-business
https://docs.python-telegram-bot.org/en/stable/telegram.businessconnection.html

**Comment by pranavrajs** [comment]
@IGNOR90 Moved this to discussions, can you please share more about the expectations here? I'm not sure if I understand the request.

**Reply by webant-ltd** [reply]
Telegram now has the ability to manage user’s personal messages through a bot
https://core.telegram.org/bots#manage-your-business

I would like to be able to have messages sent to chatwoot when connecting such a bot.

**Reply by SwimGeek** [reply]
Hi, I'm using a Telegram Business account  - and linked a bot to it.

Under bot settings I had to enable 'business mode'.

Chatwoot lets mee link the bot to an inbox. But I'm not getting any messages into Chatwoot when I send messages to the Telegram account / bot.

Please support 'business mode' Telegram bots.

**Comment by SwimGeek** [comment]
If the bot is in 'business mode' - you can chat with it directly and messages go into chatwoot.

But, if you chat with the business account (with linked bot), the messages do not show up in chatwoot.

**Reply by insoln** [reply]
The problem is that "linked" updates, unlike direct ones, have "business_message" attribute instead of a regular "message". Additionally, this "business_message" has "business_connection_id" that refers to exact linked account which received the message.

When sending a reply to the linked account the bot must specify that business_connection_id.

**Reply by SwimGeek** [reply]
Hi

Thanks for that comment - makes sense.

@pranavrajs sounds like the inbox config can remain the same - linked to a Telegram bot - but Chatwoot needs to be aware of the "business_message" attribute case.

**Reply by insoln** [reply]
If inbox config remains the same, it is required to store business_connection_id in a conversation. Also, like WABA, Telegram Business has 24-hr rolling window for replies.

While it may be tempting to have separate inbox configs for direct and linked conversations (code simplicity, opportunity to have different inboxes for different business accounts), there is one more catch: Telegram allows only one bot handler to receive updates, so multiple inboxes for the same bot (but different business accounts) have to share a single Telegram connection, which sounds tricky.

**Reply by SwimGeek** [reply]
Hi

I don't think Telegram Business has the 24 hour messaging window limit?

I noticed the single bot per Telegram Business account limit. Ideally on the Chatwoot side the same inbox should handle normal bot chats and business bot chats (for the same bot).

**Comment by Pubgwala420** [comment]
How to get business_connection_id
